x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/queries/InnerChunkBuilder.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.queries;
+
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.index.query.InnerHitBuilder;
+import org.elasticsearch.search.builder.SearchSourceBuilder;
+import org.elasticsearch.xcontent.ObjectParser;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.index.query.InnerHitBuilder.DEFAULT_FROM;
+import static org.elasticsearch.index.query.InnerHitBuilder.DEFAULT_SIZE;
+
+public class InnerChunkBuilder implements Writeable, ToXContentObject {
+    private static final ObjectParser<InnerChunkBuilder, Void> PARSER = new ObjectParser<>("inner_chunks", InnerChunkBuilder::new);
+
+    static {
+        PARSER.declareInt(InnerChunkBuilder::setFrom, SearchSourceBuilder.FROM_FIELD);
+        PARSER.declareInt(InnerChunkBuilder::setSize, SearchSourceBuilder.SIZE_FIELD);
+    }
+
+    private final String name;
+    private int from = DEFAULT_FROM;
+    private int size = DEFAULT_SIZE;
+
+    public InnerChunkBuilder() {
+        // Set hard-coded name value here so that if we change it in the future, the updated value is serialized/deserialized and propagated
```

Comment:
I'm not sure I understand why we need to do this?

Reply:
I did this to implement a hard-coded name that will still be serialized so that if we change it in the future, we have one less BwC problem in a mixed-cluster environment. But it's moot now, based on @joemcelroy's comment the name will have to be dynamic anyways.